packages/dialtone-tokens/themes/config.js

```diff
@@ -61,3 +190,341 @@ function _setThemeAttributeOnRoot (theme, brand, rootNode) {
+  if (!brandTheme || typeof brandTheme !== 'object') {
+    throw new TypeError(
+      '[Dialtone] initDialtoneTheme: brandTheme must be an object. ' +
+      'Import a brand theme like: import Dp from \'@dialpad/dialtone-tokens/themes/dp\';',
+    );
+  }
+
+  // Validation: brandTheme.brand must exist and be properly structured
+  if (!brandTheme.brand || typeof brandTheme.brand !== 'object') {
+    throw new TypeError(
+      '[Dialtone] initDialtoneTheme: brandTheme.brand must be an object with {name, css} properties. ' +
+      'Ensure you\'re importing a valid brand theme.',
+    );
+  }
+
+  if (typeof brandTheme.brand.name !== 'string' || !brandTheme.brand.name) {
+    throw new TypeError(
+      '[Dialtone] initDialtoneTheme: brandTheme.brand.name must be a non-empty string.',
+    );
+  }
+
+  if (typeof brandTheme.brand.css !== 'string') {
+    throw new TypeError(
+      '[Dialtone] initDialtoneTheme: brandTheme.brand.css must be a string containing CSS.',
+    );
+  }
+
+  // Validation: mode must be valid
+  if (mode !== 'light' && mode !== 'dark') {
+    throw new TypeError(
+      `[Dialtone] initDialtoneTheme: mode must be 'light' or 'dark', got '${mode}'.`,
+    );
+  }
+
+  // Validation: rootNode must be a valid element
+  if (!rootNode || (typeof rootNode !== 'object')) {
+    throw new TypeError(
+      '[Dialtone] initDialtoneTheme: rootNode must be an HTMLElement or host element with shadowRoot.',
+    );
+  }
+
+  // Warn if someone passed shadowRoot directly instead of the host
+  if (rootNode instanceof ShadowRoot) {
+    console.warn(
+      '[Dialtone] You passed a ShadowRoot directly to initDialtoneTheme(). ' +
+      'Please pass the host element instead. The function will access shadowRoot automatically.\n' +
+      'Correct: initDialtoneTheme(brand, mode, hostElement)\n' +
+      'Incorrect: initDialtoneTheme(brand, mode, hostElement.shadowRoot)',
+    );
+  }
+
+  if (rootNode?.shadowRoot) {
+    rootNode = rootNode.shadowRoot;
+  }
+
+  // Check for duplicate initialization (idempotency protection with soft warnings)
+  const existing = initializationState.get(rootNode);
+  if (existing) {
+    if (existing.brand === brandTheme.brand.name && existing.mode === mode) {
+      console.warn(
+        `[Dialtone] Theme already initialized for this rootNode with brand '${brandTheme.brand.name}' and mode '${mode}'. ` +
+        'Re-applying the same theme may be unnecessary. ' +
+        'If you need to switch themes dynamically, use setBrand() or setMode() instead of calling initDialtoneTheme() again.',
+      );
+    } else {
+      console.warn(
+        `[Dialtone] Theme already initialized for this rootNode. ` +
+        `Previous: brand='${existing.brand}', mode='${existing.mode}'. ` +
+        `New: brand='${brandTheme.brand.name}', mode='${mode}'. ` +
+        'Re-initializing with different parameters. Consider using setBrand()/setMode() for dynamic switching.',
+      );
+    }
+  }
```

Comment:
I don't think it's really that necessary to do this check at all, but if we do want to do it we should output the warning and then immediately `return;` to exit this function.

Reply:
Added early returns after warnings. That should ensure developers use correct patternof `setBrand()`/`setMode()` ratherthan re-calling `initDialtoneTheme()`. https://github.com/dialpad/dialtone/pull/945/commits/311e8e3589a60fa18b4d78f445658281bf599489